tests/PySys/software-management-end-to-end/sm-apt-install-remove-with-version/run.py

```diff
@@ -0,0 +1,52 @@
+from pysys.basetest import BaseTest
+
+import time
+
+"""
+Validate end to end behaviour for the apt plugin for packages with version
+
+When we install a package
+Then it is installed
+When we deinstall it again with the wrong version
+Then it is still installed
+When we deinstall it again with the right version
+Then it is not installed
+
+"""
+
+import json
+import requests
+import time
+import sys
+
+sys.path.append("software-management-end-to-end")
+from environment_sm_management import SmManagement
+
+
+class PySysTest(SmManagement):
+    def setup(self):
+        super().setup()
+        self.assertThat("False == value", value=self.check_isinstalled("rolldice"))
+
+    def execute(self):
+
+        self.trigger_action("rolldice", "5445239", "1.16-1+b1::apt", "notanurl", "install")
+
+        self.wait_until_succcess()
+
+        self.assertThat("True == value", value=self.check_isinstalled("rolldice"))
+
+        self.trigger_action("rolldice", "5445239", "88::apt", "notanurl", "delete")
+
+        self.wait_until_fail()
+
+        self.assertThat("True == value", value=self.check_isinstalled("rolldice"))
+
+        self.trigger_action("rolldice", "5445239", "1.16-1+b1::apt", "notanurl", "delete")
+
+        self.wait_until_succcess()
```

Comment:
I believe a state check after each would be nice to have in place to make sure that we are actually applying the actions.

Reply:
Hi, I hope I did this with the assertThat  and the wait_until_statements. Or do you have another state check in mind?